poetry/mixology/incompatibility.py

```diff
@@ -57,7 +57,10 @@ def __init__(self, terms: List[Term], cause: IncompatibilityCause) -> None:
                     # irrelevant, since we already know that mutually exclusive version
                     # ranges are incompatible. We should never derive an irrelevant
                     # incompatibility.
-                    assert by_ref[ref] is not None
+                    err_msg = "Package '{}' is listed as a dependency of itself.\nThis is not supported by Poetry.".format(
+                        ref
+                    )
```

Comment:
```suggestion
                    err_msg = f"Package '{ref}' is listed as a dependency of itself."
```

I'm not aware that listing itself is possible at all. That's why I would remove this sentence.

Reply:
Sure makes sense.